moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -55,46 +57,6 @@ constexpr double DEFAULT_TIMESTEP = 1e-3;
 constexpr double EPS = 1e-6;
 }  // namespace
 
-class LinearPathSegment : public PathSegment
```

Comment:
So we're promoting `LinearPathSegment` from an implementation detail to a public interface? Are we sure we want to do that and start making guarantees on its stability?